lib/ohai/system.rb

```diff
@@ -107,6 +108,21 @@ def run_plugins(safe = false, attribute_filter = nil)
         Ohai::Log.error("Encountered error while running plugins: #{e.inspect}")
         raise
       end
+      critical_failed = []
+      @runner.failed_plugins.each do |failed|
+        if Ohai::Config.ohai[:critical_plugins].include?(failed)
```

Comment:
You can use a set intersection to do this, `critical_failed = Ohai::Config.ohai[:critical_plugins] & @runner.failed_plugins`.